### PR TITLE
Use Xilinx IP in `callisto` for synthesizability

### PR DESCRIPTION
**Pull request by martijnbastiaan**
This still needs some work, mainly in the form of cautionary comments. Hence, this PR is still a draft.

### CODE REVIEW

**Comment by martijnbastiaan** [comment]
Even though it's a draft, feel free to review @vmchale !

**Comment by martijnbastiaan** [comment]
This is done, however, I want to merge it after https://github.com/bittide/bittide-hardware/pull/161 so I can move the synthesizable `callisto` to `bittide` (and prevent merge conflicts).

**Review by lmbollen — APPROVED** [review]
LGTM modulo optimizations and some small comments.

I'm still missing a test we can run on CI that verifies the correct functioning of the control algorithm. Given that we're currently prioritizing a running clock demo, I'll approve this PR so it does not hold up other PRs and I will make an issue to implement a test for CI.

File: elastic-buffer-sim/src/Bittide/ClockControl/Strategies/Callisto.hs
Comment (vmchale) [inline elastic-buffer-sim/src/Bittide/ClockControl/Strategies/Callisto.hs:51]: You can use TeX in haddock!

https://haskell-haddock.readthedocs.io/en/latest/markup.html

File: elastic-buffer-sim/src/Bittide/ClockControl/Strategies/Callisto.hs
Comment (vmchale) [inline elastic-buffer-sim/src/Bittide/ClockControl/Strategies/Callisto.hs:92]: Should this be done before merging?

Comment (martijnbastiaan) [reply elastic-buffer-sim/src/Bittide/ClockControl/Strategies/Callisto.hs:92]: No, 32 bit should be plenty for our demos. I'm trying to speedrun getting things working on the board.

File: elastic-buffer-sim/src/Bittide/ClockControl/Strategies/Callisto.hs
Comment (martijnbastiaan) [reply elastic-buffer-sim/src/Bittide/ClockControl/Strategies/Callisto.hs:37]: Agreed. I tried, but this is unfortunately more work than it seems due to the propagation of type variables throughout the rest of the code.

**Comment by martijnbastiaan** [comment]
I've added more documentation and added an explanation on what we can improve (and why we didn't). I'm postponing merging so I can get simulation on CI.

File: elastic-buffer-sim/src/Bittide/ClockControl/Strategies/Callisto.hs
Comment (martijnbastiaan) [reply elastic-buffer-sim/src/Bittide/ClockControl/Strategies/Callisto.hs:64]: I'm going to move forward with the merge, given that this isn't critical. Please make a papercut PR if you feel I should have done a different mark up style :).

**Comment by martijnbastiaan** [comment]
I'm dumb. I promised myself to defer merging this to prevent merge conflicts with #161 :facepalm:

**Pull request merged**
